plugins/serializers/splunkmetric/splunkmetric.go

```diff
@@ -66,37 +70,67 @@ func (s *serializer) createObject(metric telegraf.Metric) (metricGroup []byte, e
 	dataGroup := HECTimeSeries{}
 	var metricJson []byte
 
-	for _, field := range metric.FieldList() {
+	type CommonTags struct {
+		Time   float64
+		Host   string
+		Index  string
+		Source string
+		Fields map[string]interface{}
+	}
 
-		value, valid := verifyValue(field.Value)
+	// The tags are common to all events in this timeseries
+	commonTags := CommonTags{}
 
-		if !valid {
-			log.Printf("D! Can not parse value: %v for key: %v", field.Value, field.Key)
-			continue
-		}
+	commonObj := map[string]interface{}{}
+
+	commonObj["config:hecRouting"] = s.HecRouting
+	commonObj["config:multiMetric"] = s.SplunkmetricMultiMetric
 
-		obj := map[string]interface{}{}
-		obj["metric_name"] = metric.Name() + "." + field.Key
-		obj["_value"] = value
+	commonTags.Fields = commonObj
 
+	// Break tags out into key(n)=value(t) pairs
+	for n, t := range metric.Tags() {
+		if n == "host" {
+			commonTags.Host = t
+		} else if n == "index" {
+			commonTags.Index = t
+		} else if n == "source" {
+			commonTags.Source = t
+		} else {
+			commonTags.Fields[n] = t
+		}
+	}
+	commonTags.Time = float64(metric.Time().UnixNano()) / float64(1000000000)
+	switch s.SplunkmetricMultiMetric {
```

Comment:
I suggest having two functions, one for multi metric and one for single.  The idea being to keep these tasks as separate as possible and reduce the function length so that it can be more easily understood.

Reply:
done